### PR TITLE
Add input for custom cash donation

### PR DESCRIPTION
**Pull request by Tokikko**

### CODE REVIEW

**Review by gnorbsl — APPROVED** [review]

File: client/public/clientJS/stripe.js [hunk 1]
```diff
@@ -123,6 +123,26 @@ let addCashDonation = function (originalAmount, amount) {
   userDonation.innerHTML = '$' + Math.floor(amount* 100) / 100;
 }
 
+let showCustomAmountInput = () => {
+  let inputAmountElement = document.getElementById("customAmountValue");
+  const isVisible = inputAmountElement.style.visibility === 'hidden';
+  // if users hides the input reset total amount to 0
+  if (isVisible === false) {
```
Comment (gnorbsl) [inline client/public/clientJS/stripe.js:130]: if !isVisible?

File: client/public/clientJS/stripe.js [hunk 2]
```diff
@@ -123,6 +123,26 @@ let addCashDonation = function (originalAmount, amount) {
...
+    let totalAmount = document.getElementById('total-cost-hidden');
+    let parsedTotalAmount = parseFloat(totalAmount.innerText);
+    inputAmountElement.value = "";
+    addCashDonation(parsedTotalAmount, 0);
+  }
+  inputAmountElement.style.visibility = isVisible ? 'visible' : 'hidden';
+}
+
+let addCashDonationFromCustomAmountInput = (originalAmount, e) => {
+  let amount = parseInt(e.value);
```
Comment (gnorbsl) [inline client/public/clientJS/stripe.js:140]: better use float here

Comment (gnorbsl) [reply client/public/clientJS/stripe.js:140]: in case someone wants to donate pennies :D

**Review by gnorbsl — CHANGES_REQUESTED** [review]

File: client/public/clientJS/stripe.js (hunk 1 above)
Comment (Tokikko) [reply client/public/clientJS/stripe.js:130]: i prefer explicit check in cases when comparing to negatives

Comment (gnorbsl) [reply client/public/clientJS/stripe.js:130]: why, whats the difference between your code and !?

**Pull request merged**